Change Log Entries - starting from 2021-03-09 (#621)
stitchdata/docs: _changelog/entries/2021/2021-03-12-pendo-v1-ijson-performance-enhancement.md

---

> title: Pendo (v1) improvement: using ijson to improve `visitors` and `events` replication

Reviewer: ```suggestion
title: "Pendo (v1) improvement: Improve performance for`visitors` and `events` tables"
```

My thought here (aside from the double quotes) was to make this general and relatable. Mentioning how this improvement works is good in the body, but here it might not have as much impact if someone is skimming.

content-type: "changelog-entry"
date: 2021-03-12
entry-type: improvement
entry-category: "integration"

> connection-id: "pendo"
> connection-version: "1"

Reviewer: ```suggestion
connection-id: "pendo"
connection-version: "1"
pull-request: "https://github.com/singer-io/tap-pendo/pull/28"
```

---

{{ site.data.changelog.metadata.single-integration | flatify }}

> ijson is now being used for {{ this-connection.display_name }}'s `visitors` and `events` streams to improve performance. It reduces the amount of memory needed to replicate its potentially large amounts of data.

Reviewer: ```suggestion
We've implemented [ijson](https://github.com/ICRAR/ijson), which is used in other Singer taps, to streamline replication for {{ this-connection.display_name }}'s `visitors` and `events` tables. This improvement reduces the amount of memory needed to replicate large amounts of data, potentially leading to improved Extraction times.
```

I made this a bit friendlier-sounding and tried to convey why this change is good for the end user. I like it when changelogs include pronouns and personality and not just technical mumbo jumbo - it reminds the end users that there are people working on the product. I see it as us talking to them every week and chatting about what we've been doing.
